### DISCUSSION TITLE
Error while trying to install

### DISCUSSION THREAD

**Discussion by kernow** [BJReplay/ha-solcast-solar#39] [General]
Hi,

I'm switching over from the old repo and can't get this new install to work. I removed the old integration from HACS and the integration from settings. I then followed the install instructions in the README. But when I try to add the integration is says id failed to setup in the UI. I've turned on debug logging but that doesn't show any further info. The only entry in the logs that relates is this:

`Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 444, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/__init__.py", line 76, in async_setup_entry
    hass.config_entries.async_update_entry(entry, options=new, version=7)
TypeError: ConfigEntries.async_update_entry() got an unexpected keyword argument 'version'`

Anyone know how to solve this?

Thanks

**Comment by gcoan** [comment]
You didn't actually need to remove solcast from the integrations, just removing from HACS and then adding the new repository should be enough 

But anyway,  have you tried restarting home assistant after you've removed the integration?  There may be residual bits of the integration still running

**Reply by kernow** [reply]
> You didn't actually need to remove solcast from the integrations, just removing from HACS and then adding the new repository should be enough

I discovered that after it was too late unfortunately.


> But anyway, have you tried restarting home assistant after you've removed the integration? There may be residual bits of the integration still running

Yes, I have restarted several times. I've also completely removed the from HACS and reinstalled, with restarts and still the same issue.

**Reply by gcoan** [reply]
Could try deleting the solcast custom component directory but sounds like there's a dépendance on needing a newer version of HA.

I tend to leave versions a while to stabilisé, so am on 2024.6.2 and it's working fine

**Comment by autoSteve** [comment]
Have you seen #40?

There probably lies an answer. Upgrade HA.

Which version are you on? 2024.2.something? If so you're in a cohort of 3.17%.

**Reply by kernow** [reply]
I saw that thanks and replied. I'm on 2024.2.5

**Reply by autoSteve** [reply]
Then there be dragons. Upgrade, as oziee introduced a breaking change without telling anyone. And I blindly followed his work.
